File: Project/TeamA_Enterprise-Learning-Platform-main (1)/TeamA_Enterprise-Learning-Platform-main/skillsphere-frontend/src/services/InternshipService.js
import apiClient from '../api/apiClient';

const InternshipService = {
  getAllActiveInternships() {
    return apiClient.get('/api/internships');
  },

  getInternshipById(id) {
    return apiClient.get(`/api/internships/${id}`);
  },

  applyForInternship(id, applicationData) {
    return apiClient.post(`/api/internships/${id}/apply`, applicationData);
  },

  getMyApplications() {
    return apiClient.get('/api/internships/my-applications');
  },

  createInternship(data) {
    return apiClient.post('/api/internships', data);
  },

  getMentorApplications() {
    return apiClient.get('/api/internships/mentor/applications');
  },

  updateApplicationStatus(id, status, reviewNotes) {
    return apiClient.patch(`/api/internships/applications/${id}/status`, { status, reviewNotes });
  }
};

export default InternshipService;

File: Project/TeamA_Enterprise-Learning-Platform-main (1)/TeamA_Enterprise-Learning-Platform-main/skillsphere-backend/src/main/java/com/skillsphere/service/InternshipService.java
package com.skillsphere.service;

import com.skillsphere.entity.Internship;
import com.skillsphere.entity.InternshipApplication;
import com.skillsphere.entity.User;
import java.util.List;

public interface InternshipService {
    List<Internship> getAllActiveInternships();
    List<Internship> getAllInternshipsAdmin();
    Internship getInternshipById(Long id);
    Internship createInternship(Internship internship, User poster);
    InternshipApplication applyForInternship(Long internshipId, String userEmail, InternshipApplication appDetails);
    List<InternshipApplication> getStudentApplications(String userEmail);
    List<InternshipApplication> getMentorApplications(String mentorEmail);
    List<InternshipApplication> getAllApplicationsAdmin();
    InternshipApplication updateApplicationStatus(Long applicationId, String status, String reviewNotes);
}
